### ISSUE TITLE
Not maintained since Mar 14 2019

### ISSUE THREAD

**Issue by erhuz** [hubertnnn/laravel-nova-field-dynamic-select#11]

**Comment by royduin** [comment]
Yes, also no response on my pull requests.  You can use my fork: https://github.com/royduin/laravel-nova-field-dynamic-select which includes multiple fixes. Use `dev-master` with Composer.

**Comment by erhuz** [comment]
@royduin Thank you!

**Comment by mavericksthinker** [comment]
@royduin I am using your fork, but seems like the implementation is not usable in action fields as it checks for resource form fields. Any plans of providing support for that?

**Comment by royduin** [comment]
Using it myself in an action without any problems. But you could create a detailed issue in my repo or a PR with a fix if it's a bug.

**Comment by mavericksthinker** [comment]
@royduin When same field in action exists on the resource it will work fine. 
```
$fields = method_exists($resource, 'parentUpdateFields')
            ? $resource->parentUpdateFields($request)
            : $resource->updateFields($request);
```
Otherwise it wont

**Comment by royduin** [comment]
That code isn't in this package... but it looks like you're using https://github.com/eminiarts/nova-tabs which does have a `parentUpdateFields` method. Give it a try there 👍or better; fix it and create a PR.

**Comment by N1Prop** [comment]
@royduin I'm using your fork. Got an issue and I would appreciate your help. Do you know why I don't get depended value in my select? 

Here is what i have. Second select $values var is empty on any year selection. 

```
DynamicSelect::make('Year', 'year') ->options(['2020' => 2020, '2019' => 2019])

DynamicSelect::make('Make', 'make')
                ->dependsOn(['year'])
                ->options(function($value) {
                     return $this->getMakesForYear($values); // here $values are empty
 })
```

Oddly this only happens on create new. On update it works like a charm.

**Comment by royduin** [comment]
You're passing `$value` in the closure, not `$values`...

**Comment by N1Prop** [comment]
@royduin It was a typo on my end above. I'm actually passing $values and at I mentioned, why I update a resource ti works fine. It is happening only when I'm creating new. Any ideas where I should be looking at?

**Comment by royduin** [comment]
Debug it and if there is a bug fix it and create a PR or let me know though a new issue what's wrong 👍I'm not a helpdesk.

**Comment by ragingdave** [comment]
@royduin Thanks for getting this package actually working! Would you mind making an actual release based on your fork? I'm planning on using it in a production environment...and using dev-master....feels...well bad.

**Comment by royduin** [comment]
Done: https://github.com/royduin/laravel-nova-field-dynamic-select/releases

**Comment by jesperbjoernandersen** [comment]
You didn't upload it to packagist for composer installs?

**Comment by royduin** [comment]
It's now on packagist: https://packagist.org/packages/royduin/laravel-nova-field-dynamic-select 👍

**Comment by jesperbjoernandersen** [comment]
Cheers 👍

**Comment by BnitoBzh** [comment]
@royduin the "Issues" tab is not enabled on your fork ..
There is an issue with Nova >= 3.9.0, dependsOn seems to not work.
The parent field value is not available is the child field, see below.

```php
            DynamicSelect::make('Team')
                ->options(Team::all()->mapWithKeys(function (Team $team) {
                    return [$team->id => $team->name];
                }))
                ->rules('required'),

            DynamicSelect::make('User', 'user')
                ->dependsOn(['team'])
                ->options(function ($values) {
                    // ========>>> $values is empty
                    $team = Team::query()->find($values['team']);

                    if (! $team) {
                        return new Collection();
                    }

                    return $team->users()
                        ->get()
                        ->add($team->user)
                        ->mapWithKeys(function (User $user) {
                            return [$user->id => $user->name];
                        });
                })
                ->rules('required'),
```

**Comment by royduin** [comment]
Enabled it 👍 did you debug it? A pull request is welcome.

**Comment by ragingdave** [comment]
@hubertnnn can you mark your package as archived and not-maintained? It would be good for people to know there is an actually maintained version going forward.

**Comment by BnitoBzh** [comment]
> Enabled it 👍 did you debug it? A pull request is welcome.

Sorry for the late answer, i am actually very busy, if someone can debug it for us.

**Comment by BnitoBzh** [comment]
@royduin the fix : https://github.com/royduin/laravel-nova-field-dynamic-select/pull/2
